File: decompiled_melonDS/sources/me/magnum/melonds/ui/settings/fragments/AudioPreferencesFragment.java
package me.magnum.melonds.ui.settings.fragments;

import android.os.Bundle;
import androidx.fragment.app.b1;
import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.SeekBarPreference;
import me.magnum.melonds.R;
/* compiled from: r8-map-id-68a076d6c8e19c01d0388d4729d627070c4f7cff1f6777b7d5e4497c81f45b39 */
/* loaded from: classes.dex */
public final class AudioPreferencesFragment extends BasePreferenceFragment implements ci.l {
    public ListPreference A;
    public final h.c B;

    public AudioPreferencesFragment() {
        h.c registerForActivityResult = registerForActivityResult(new b1(3), new f(this, 2));
        registerForActivityResult.getClass();
        this.B = registerForActivityResult;
    }

    @Override // ci.l
    public final String getTitle() {
        String string = getString(R.string.category_audio);
        string.getClass();
        return string;
    }

    public final void h(boolean z10) {
        if (!z10 && shouldShowRequestPermissionRationale("android.permission.RECORD_AUDIO")) {
            bk.a aVar = new bk.a(requireContext());
            aVar.w(R.string.microphone_permission_required);
            aVar.t(R.string.microphone_permission_required_info);
            aVar.v(R.string.ok, new b(this, 0));
            aVar.y();
            return;
        }
        this.B.a("android.permission.RECORD_AUDIO");
    }

    @Override // androidx.preference.a0
    public final void onCreatePreferences(Bundle bundle, String str) {
        setPreferencesFromResource(R.xml.pref_audio, str);
        Preference findPreference = findPreference("volume");
        findPreference.getClass();
        SeekBarPreference seekBarPreference = (SeekBarPreference) findPreference;
        Preference findPreference2 = findPreference("mic_source");
        findPreference2.getClass();
        this.A = (ListPreference) findPreference2;
        seekBarPreference.setSummary(getString(R.string.volume_percentage, Integer.valueOf((int) ((seekBarPreference.A / seekBarPreference.L) * 100.0f))));
        seekBarPreference.setOnPreferenceChangeListener(new ci.h(this, seekBarPreference, 2));
        ListPreference listPreference = this.A;
        if (listPreference != null) {
            listPreference.setOnPreferenceChangeListener(new a(0, this));
        } else {
            nc.k.f("micSourcePreference");
            throw null;
        }
    }
}

File: app/src/main/java/me/magnum/melonds/ui/settings/fragments/AudioPreferencesFragment.kt
package me.magnum.melonds.ui.settings.fragments

import android.Manifest
import android.os.Bundle
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AlertDialog
import androidx.preference.ListPreference
import androidx.preference.SeekBarPreference
import dagger.hilt.android.AndroidEntryPoint
import me.magnum.melonds.R
import me.magnum.melonds.domain.model.MicSource
import me.magnum.melonds.extensions.isMicrophonePermissionGranted
import me.magnum.melonds.ui.settings.PreferenceFragmentTitleProvider
import me.magnum.melonds.utils.enumValueOfIgnoreCase

@AndroidEntryPoint
class AudioPreferencesFragment : BasePreferenceFragment(), PreferenceFragmentTitleProvider {

    private var micSourcePreference: ListPreference? = null

    private val microphonePermissionLauncher = registerForActivityResult(ActivityResultContracts.RequestPermission()) { granted ->
        if (granted) {
            micSourcePreference?.value = MicSource.DEVICE.name.lowercase()
        }
    }

    override fun getTitle() = getString(R.string.category_audio)

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.pref_audio, rootKey)
        val volumePreference = findPreference<SeekBarPreference>("volume")
        micSourcePreference = findPreference("mic_source")

        volumePreference?.let {
            updateVolumePreferenceSummary(it, it.value)
            it.setOnPreferenceChangeListener { _, newValue ->
                updateVolumePreferenceSummary(it, newValue as Int)
                true
            }
        }

        micSourcePreference?.setOnPreferenceChangeListener { _, newValue ->
            val newMicSource = enumValueOfIgnoreCase<MicSource>(newValue as String)
            if (newMicSource == MicSource.DEVICE && !requireContext().isMicrophonePermissionGranted()) {
                requestMicrophonePermission(false)
                false
            } else {
                true
            }
        }

        hideDependentsWhenInactive("sound_enabled", "audio_interpolation", "audio_bitrate", "volume", "audio_latency", "audio_soft_limiter_enabled", "audio_bass_boost_enabled", "audio_bass_boost_strength", "audio_spatial_audio_enabled", "audio_reverb_enabled")
    }

    private fun updateVolumePreferenceSummary(volumePreference: SeekBarPreference, volume: Int) {
        val volumePercentage = (volume / volumePreference.max.toFloat() * 100f).toInt()
        volumePreference.summary = getString(R.string.volume_percentage, volumePercentage)
    }

    private fun requestMicrophonePermission(overrideRationaleRequest: Boolean) {
        if (!overrideRationaleRequest && shouldShowRequestPermissionRationale(Manifest.permission.RECORD_AUDIO)) {
            AlertDialog.Builder(requireContext())
                .setTitle(R.string.microphone_permission_required)
                .setMessage(R.string.microphone_permission_required_info)
                .setPositiveButton(R.string.ok) { _, _ -> requestMicrophonePermission(true) }
                .show()
        } else {
            microphonePermissionLauncher.launch(Manifest.permission.RECORD_AUDIO)
        }
    }
}